### PR TITLE
Integrate moin cluster into scs monitoring

### LINKED ISSUE #32: Integrate `moin.k8s.scs.community` cluster into our monitoring stack https://monitoring.scs.community
We want to integrate moin cluster into our already existing SCS monitoring, basically want to achieve the one new panel under the `Cluster` row at the main overview page

Related to #18 

- [x] enable metrics
- [x] enable logs

### PR DESCRIPTION
**Pull request by michal-gubricky**
This PR integrates moin cluster into our already existing SCS monitoring

Fixes #32 
Related https://github.com/SovereignCloudStack/k8s-observability/pull/65

### CODE REVIEW

**Comment by michal-gubricky** [comment]
This PR does not yet contain Loki, only metrics for now.

File: prod/monitoring/installation/values-workload.yaml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        replicaCount: 2
+        replicaLabel: [prometheus_replica]
+    ## FIXME: Thanos query is temporary exposed via (unsecure) NodePort, see https://github.com/SovereignCloudStack/k8s-observability/issues/32
+    ##   Thanos query should be exposed via ingress, see below commented section
+        serviceGrpc:
+          type: NodePort
+          nodePorts:
+            grpc: "31050"
+
+    containerLabel: &containerLabel
```
Comment (chess-knight) [inline prod/monitoring/installation/values-workload.yaml:35]: Where is this used? Do you miss the `dnation-kubernetes-monitoring` section here? See https://github.com/SovereignCloudStack/k8s-observability/blob/main/registry/workload/values-workload.yaml#L39

File: prod/monitoring/installation/helmrelease.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+apiVersion: helm.toolkit.fluxcd.io/v2beta1
+kind: HelmRelease
+metadata:
+  name: dnation-kubernetes-monitoring-stack
+  namespace: flux-system
+spec:
+  releaseName: dnation-kubernetes-monitoring-stack
+  chart:
+    spec:
+      chart: dnation-kubernetes-monitoring-stack
```
Comment (chess-knight) [inline prod/monitoring/installation/helmrelease.yaml:11]: I think we should pin the version here.

File: prod/monitoring/installation/helmrepo.yaml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+---
+apiVersion: source.toolkit.fluxcd.io/v1beta2
+kind: HelmRepository
+metadata:
+  name: dnation-cloud
+  namespace: flux-system
+spec:
+  url: https://charts.dnationcloud.com/
```
Comment (chess-knight) [inline prod/monitoring/installation/helmrepo.yaml:8]: Is this URL correct?

File: prod/monitoring/installation/helmrelease.yaml (hunk 2 above)
Comment (michal-gubricky) [reply prod/monitoring/installation/helmrelease.yaml:11]: Yes, we can do that. I would propose to pin the latest version which is `3.6.1`.

File: prod/monitoring/installation/values-workload.yaml (hunk 1 above)
Comment (michal-gubricky) [reply prod/monitoring/installation/values-workload.yaml:35]: `containerLabel` field is unnecessary for now and can be deleted.

Comment (chess-knight) [reply prod/monitoring/installation/values-workload.yaml]: And what about `dnation-kubernetes-monitoring` section?

Comment (chess-knight) [reply prod/monitoring/installation/values-workload.yaml]: Ok, I see. Thanks

File: prod/flux/config/prod-monitoring-installation-ks.yaml
Comment (mxmxchere) [inline prod/flux/config/prod-monitoring-installation-ks.yaml:1]: We have to add this file to `kustomization.yaml` in the same directory.

**Comment by mxmxchere** [comment]
Hi michal, thanks for your nice PR, i would like to make a seperate PR for adding your pgp-keys, i propose the following approach:

1. you make a new "keys"-PR with your key in hack/pgp_keys/mgubricky.pub.asc and your fingerprint in 75676F49EAFAA9CA .sops.yaml
2. i reencrypt all secrets that currently exist and push the changes to the "keys"-PR, afterwards we merge
3. we rebase the this PR and you can encrypt the secret with all keys, (i will figure out the command and send it to you when i have it :D)
I suggest we add @chess-knight key in the same run, what do you think?

apart from that i like you PR, you picked up the existing structure quite nicely, that gives me the feeling that the folder structure gives new people a chance to get around.

File: prod/flux/config/prod-monitoring-installation-ks.yaml
Comment (michal-gubricky) [reply prod/flux/config/prod-monitoring-installation-ks.yaml:1]: Sure, I overlooked it somehow. Added in 284b1e54816056b244181c96ded298216a2f4607

**Comment by michal-gubricky** [comment]
> Hi michal, thanks for your nice PR, i would like to make a seperate PR for adding your pgp-keys, i propose the following approach:
> 
> 1. you make a new "keys"-PR with your key in hack/pgp_keys/mgubricky.pub.asc and your fingerprint in 75676F49EAFAA9CA .sops.yaml
> 2. i reencrypt all secrets that currently exist and push the changes to the "keys"-PR, afterwards we merge
> 3. we rebase the this PR and you can encrypt the secret with all keys, (i will figure out the command and send it to you when i have it :D)
>    I suggest we add @chess-knight key in the same run, what do you think?
> 
> apart from that i like you PR, you picked up the existing structure quite nicely, that gives me the feeling that the folder structure gives new people a chance to get around.

Hi Malte,

Sure, we can do that in a separate PR as well. I will create a new PR to add my key and, if @chess-knight wants, his key too. Yes, the folder structure is understandable, and as a new person, I had no problem understanding it.

**Comment by mxmxchere** [comment]
Looks good so far, in order to test the state i propose to renconcile the cluster on the state of this PR. We can then fix things that do not work well and merge a good state to main later. 

In order to do that we have to set the reconciling target  (https://github.com/SovereignCloudStack/moin-cluster-config/blob/e2e058b41e116a8903b7c414079da1ecfc0746a9/flux/config/flux-config-ks.yaml#L10) in this PR to the branch of this PR itself (otherwise the cluster will jump back to main).



The procedure to merge back to main when we are finished fixing will be the following
1. suspend flux gitrepo resource
2. set target (https://github.com/SovereignCloudStack/moin-cluster-config/blob/e2e058b41e116a8903b7c414079da1ecfc0746a9/flux/config/flux-config-ks.yaml#L10) in this PR to main
3. merge
4. set branch in flux gitrepo resource manually to main (via k edit)
5. resume gitrepo

**Comment by michal-gubricky** [comment]
> Looks good so far, in order to test the state i propose to renconcile the cluster on the state of this PR. We can then fix things that do not work well and merge a good state to main later.
> 
> In order to do that we have to set the reconciling target (
> 
> https://github.com/SovereignCloudStack/moin-cluster-config/blob/e2e058b41e116a8903b7c414079da1ecfc0746a9/flux/config/flux-config-ks.yaml#L10
> 
> ) in this PR to the branch of this PR itself (otherwise the cluster will jump back to main).
> The procedure to merge back to main when we are finished fixing will be the following
> 
> 1. suspend flux gitrepo resource
> 2. set target (https://github.com/SovereignCloudStack/moin-cluster-config/blob/e2e058b41e116a8903b7c414079da1ecfc0746a9/flux/config/flux-config-ks.yaml#L10
>    ) in this PR to main
> 3. merge
> 4. set branch in flux gitrepo resource manually to main (via k edit)
> 5. resume gitrepo

It seems like the integration of the Moin cluster into the SCS monitoring is working. The Moin cluster can be seen at https://monitoring.scs.community/. Some manual steps need to be done on control plane components to expose metrics from them.

**Comment by michal-gubricky** [comment]
To expose the metrics of the control plane components, it was necessary to connect to each master node and edit the manifests of these components as described in the documentation. Please refer to the [docs](https://dnationcloud.github.io/kubernetes-monitoring/helpers/FAQ/#kubernetes-monitoring-shows-or-0-state-for-some-control-plane-components-are-control-plane-components-working-correctly) for details.

**Comment by mxmxchere** [comment]
manual changes to the controlplane components will get lost when a new node is created by CAPI. This will happen with the next update for example. But this is nothing we can fix in the context of this repo.

**Comment by michal-gubricky** [comment]
> manual changes to the controlplane components will get lost when a new node is created by CAPI. This will happen with the next update for example. But this is nothing we can fix in the context of this repo.

Sure, I know that. These manual changes were the fastest way to test that monitoring is working correctly for the control plane components as well.

**Comment by michal-gubricky** [comment]
Logs from the moin-cluster should now be also available in SCS monitoring under a label `moni-cluster`.

**Review by chess-knight — APPROVED** [review]

**Pull request merged**